### PR TITLE
:pencil2: fix: issue no.15159

### LINKED ISSUE #15159: Use a newer version of `mypy==0.971`
## Proposed change
Apply the following patch (as of now) to `requirements/pytorch/test.txt` just like #15158 and fix typing following the raised errors:
```diff
-mypy==0.971
+mypy==0.982
```

### Motivation
To stay up-to-date :)

### Additional context
As we've seen in https://github.com/Lightning-AI/lightning/pull/15158#issuecomment-1281698911, the new mypy 0.982 raises around 40 errors in our codebase.

<details>
<summary>click here for mypy errors</summary>

```
Found 40 errors in 26 files (checked 417 source files)
src/lightning_lite/lite.py:172: error: Unused "type: ignore" comment
src/pytorch_lightning/profilers/pytorch.py:345: error: Incompatible return value type (implicitly returns "None", expected "Union[int, float]")  [return-value]
src/pytorch_lightning/core/saving.py:146: error: Incompatible return value type (implicitly returns "None", expected "Union[LightningModule, LightningDataModule]")  [return-value]
src/pytorch_lightning/trainer/connectors/logger_connector/result.py:228: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/epoch/prediction_epoch_loop.py:50: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/epoch/prediction_epoch_loop.py:70: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:73: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:105: error: Unused "type: ignore" comment
src/pytorch_lightning/utilities/logger.py:91: error: Incompatible return value type (implicitly returns "None", expected "List[Any]")  [return-value]
src/pytorch_lightning/tuner/lr_finder.py:104: error: Incompatible return value type (implicitly returns "None", expected "Callable[[Trainer], None]")  [return-value]
src/pytorch_lightning/loggers/wandb.py:365: error: Unused "type: ignore" comment
src/pytorch_lightning/loggers/tensorboard.py:156: error: Unused "type: ignore" comment
src/pytorch_lightning/loggers/neptune.py:323: error: Unused "type: ignore" comment
src/pytorch_lightning/loggers/neptune.py:353: error: Unused "type: ignore" comment
src/pytorch_lightning/loggers/mlflow.py:155: error: Unused "type: ignore" comment
src/pytorch_lightning/loggers/csv_logs.py:172: error: Unused "type: ignore" comment
src/pytorch_lightning/loggers/comet.py:268: error: Unused "type: ignore" comment
src/pytorch_lightning/core/module.py:602: error: Incompatible return value type (implicitly returns "None", expected "Union[Tensor, Dict[str, Any]]")  [return-value]
src/pytorch_lightning/callbacks/progress/rich_progress.py:380: error: Incompatible return value type (implicitly returns "None", expected "TaskID")  [return-value]
src/pytorch_lightning/loops/fit_loop.py:200: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/fit_loop.py:221: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/fit_loop.py:253: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/optimization/optimizer_loop.py:191: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/optimization/optimizer_loop.py:198: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/optimization/manual_loop.py:101: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/epoch/training_epoch_loop.py:1[59](https://github.com/Lightning-AI/lightning/actions/runs/3269716426/jobs/5377534254#step:6:60): error: Unused "type: ignore" comment
src/pytorch_lightning/loops/epoch/training_epoch_loop.py:174: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/dataloader/prediction_loop.py:84: error: Unused "type: ignore" comment
src/pytorch_lightning/loops/batch/training_batch_loop.py:[61](https://github.com/Lightning-AI/lightning/actions/runs/3269716426/jobs/5377534254#step:6:62): error: Unused "type: ignore" comment
src/pytorch_lightning/loops/batch/training_batch_loop.py:[70](https://github.com/Lightning-AI/lightning/actions/runs/3269716426/jobs/5377534254#step:6:71): error: Unused "type: ignore" comment
src/pytorch_lightning/lite/lite.py:1[79](https://github.com/Lightning-AI/lightning/actions/runs/3269716426/jobs/5377534254#step:6:80): error: Incompatible return value type (implicitly returns "None", expected "Strategy")  [return-value]
src/pytorch_lightning/demos/boring_classes.py:98: error: Unused "type: ignore" comment
src/pytorch_lightning/demos/boring_classes.py:110: error: Unused "type: ignore" comment
src/pytorch_lightning/demos/boring_classes.py:122: error: Unused "type: ignore" comment
src/pytorch_lightning/demos/boring_classes.py:131: error: Unused "type: ignore" comment
src/pytorch_lightning/demos/boring_classes.py:197: error: Unused "type: ignore" comment
src/pytorch_lightning/demos/boring_classes.py:214: error: Unused "type: ignore" comment
src/pytorch_lightning/demos/boring_classes.py:217: error: Unused "type: ignore" comment
```

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

### PR DESCRIPTION
**Pull request by MohamadAlghish93**
upgrade lib version

## What does this PR do?

Fixes #15159

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by awaelchli** [comment]
@MohamadAlghish93 Thanks for looking into it. Someone else is already working on this, see https://github.com/Lightning-AI/lightning/pull/15161, so I'm closing this now.
